src/app/delegation_backend/src/itn_uptime_analyzer/time.go

```diff
@@ -0,0 +1,121 @@
+package itn_uptime_analyzer
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	sheets "google.golang.org/api/sheets/v4"
+)
+
+// Returns current time in UTC format
+func GetCurrentTime() time.Time {
+	currentTime := time.Now().UTC()
+	return currentTime
+}
+
+// Checks the spreadsheet for a possible value for the last execution date
+// If there is no date, or the date is between the +/- one hour range of the chosen execution interval then the function returns current time minus the execution interval
+func GetLastExecutionTime(config AppConfig, client *sheets.Service, log *logging.ZapEventLogger, sheetTitle string, currentTime time.Time, executionInterval int) time.Time {
+	readRange := fmt.Sprintf("%s!A%d:Z%d", sheetTitle, 1, 1)
+	spId := config.AnalyzerOutputGsheetId
+
+	resp, err := client.Spreadsheets.Values.Get(spId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v\n", err)
+	}
+
+	var lastFilledColumn int = len(resp.Values[0]) - 1
+	var lastExecutionBasedOnSheetsAsTime time.Time
+
+	readRange = fmt.Sprintf("%s!%s%d", sheetTitle, string(lastFilledColumn+65), 1)
+
+	lastTimeWindow := resp.Values[0][lastFilledColumn]
+
+	stringSplit := strings.SplitAfter(fmt.Sprint(lastTimeWindow), " - ")
+	lastExecutionBasedOnSheets := stringSplit[len(stringSplit)-1]
+
+	if !strings.HasPrefix(lastExecutionBasedOnSheets, "Node") {
```

Comment:
A few things here:

1. I'd suggest to have `if strings.HasPrefix(lastExecutionBasedOnSheets, "Node")` only. Given  it returns, no need to have other case within `if` or `else`
2. Could you add a comment on why some columns may contain time and others a `Node` prefix? Completely unclear.

Reply:
Okay, I get it why it might be `Node`, but maybe it could be just `lastExecutionBasedOnSheets == "Node"` if it's just a column name?

And a comment would still be valuable to have in code

Reply:
It cannot be just `lastExecutionBasedOnSheets == "Node"` because the A1 cell value is the one we are trying to avoid. On every sheet the text is: `Node Identity ↓ Execution Time Window →` . I just used `hasPrefix` to avoid dealing with special characters like the arrows